include/envoy/network/listener.h

```diff
@@ -114,6 +114,46 @@ class ListenerCallbacks {
   virtual void onNewConnection(ConnectionPtr&& new_connection) PURE;
 };
 
+/**
+ * UDP listener callbacks.
+ */
+class UdpListenerCallbacks {
```

Comment:
I thought Matt requested on the prior PR that we try to stick to listenercallbacks where possible?

I'd think the current code could be a subclass of ListenerCallbacks, where onAccept was commeted as "accepting new data" rather than "accepting a new connection", we call onNewConnection where we currently do (but with a UdpConnection subclass which may be filled with TODOs, but we pass the local address and the peer address to it) and then hand onData calls to the new connection?

Reply:
Same comment as below. The current `ListenerCallbacks` interface exposes connection concept. Inheriting that interface would semantically mean - `UdpListenerCallbacks` `is-a` `ListenerCallbacks`. Not sure if we mean that. What about having a `bridge` that bridges the UDP listener to the existing `ListenerCallbacks`? This bridge could be a concrete implementation of a `UdpListenerCallbacks` interface.  Thinking out loud here. Would he happy to hear what you think.

Reply:
See my comments below.